lib/std/Build.zig

```diff
@@ -1928,6 +1928,51 @@ pub const LazyPath = union(enum) {
             .generated => |gen| .{ .generated = gen },
         };
     }
+
+    /// Returns a `LazyPath` that points to the directory name of the current `LazyPath`.
+    pub fn dirname(path: LazyPath) LazyPath {
+        const ComputeStep = struct {
+            step: Step,
+            input: LazyPath,
+            output: GeneratedFile,
+
+            fn make(comp_step: *Step, progress: *std.Progress.Node) !void {
+                _ = progress;
+
+                const self = @fieldParentPtr(@This(), "step", comp_step);
+
+                const realpath = self.input.getPath2(comp_step.owner, comp_step);
+
+                self.output.path = resolve(realpath);
+            }
+
+            fn resolve(realpath: []const u8) []const u8 {
+                return std.fs.path.dirname(realpath) orelse if (std.fs.path.isAbsolute(realpath)) "/" else ".";
```

Comment:
On WIndows, we will blindly return `/` as root for an abs path which is wrong.

Reply:
> On WIndows, we will blindly return `/` as root for an abs path which is wrong.

Actually not quite sure, windows also allows current-drive absolute paths that don't use a drive specifier (`\Windows\system32\calc.exe`) is valid if your CWD is in `C:\`. I gotta verify that, will test when i have a windows machine at hand

Reply:
Either way, whatever the actual drive or path notation is, we should return that instead of `/`.

Reply:
After thinking about it for a while, i think the correct solution here is one of:
1. Panic and let the user fix it
2. Pass in a builder, resolve the paths to absolute paths and then invoke dirname on those (using `.` is also a bad idea here)
3. Rewrite the Build to allow using `Build.relPath(…)`, `Build.cwdPath(…)` which both resolve to the correct path internally. This is probably the correct solution here, as `.path` is currently broken-by-design:  
What will happen if a dependency Build will pass a `.{ .path = "local" }` to a "higher" builder? It will suddenly point to a file in the local folder instead of the one where the dependency was fetched to. 

cc @andrewrk